src/main/java/core/model/HOParameter.java

```diff
@@ -52,8 +51,8 @@ public static HOParameter instance() {
     public int HOUsers = 0;
 
 	@Override
-	public HashMap<String,String> getValues() {
-		HashMap<String,String> map = new HashMap<String,String>();
+	public Map<String,String> getValues() {
+		Map<String,String> map = new HashMap<>();
```

Comment:
Here is a good illustration of what I meant earlier: if we ever decide to use a `TreeMap` to always get sorted keys, you would only change code here, the rest would remain the same: it would still handle a `Map`.